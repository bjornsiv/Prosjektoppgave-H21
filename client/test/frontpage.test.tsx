import * as React from 'react';
import mockReviewService from '../__mocks__/mock-review-service';
import mockGameService from '../__mocks__/mock-game-service';
import renderer from 'react-test-renderer';
import FrontPage from '../src/frontpage';
import { HashRouter } from 'react-router-dom';

jest.mock('../src/game-service', () => {
  return new mockGameService();
});
jest.mock('../src/review-service', () => {
  return new mockReviewService();
});

describe('Game review component tests', () => {
  test('GameReview renders correctly with snapshot', () => {
    const tree = renderer
      .create(<HashRouter><FrontPage/></HashRouter>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});