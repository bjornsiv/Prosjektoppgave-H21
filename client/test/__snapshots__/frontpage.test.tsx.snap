// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Game review component tests GameReview renders correctly with snapshot 1`] = `
<div>
  <div>
    <div
      className="card rounded"
    >
      <div
        className="card-body"
      >
        <p
          className="card-title"
        >
          Popular games right now
        </p>
        <div
          className="card-text"
        >
          <br />
          <div
            className="row"
          >
            <div
              className="col"
            >
              <div
                className="float-start"
              />
            </div>
            <div
              className="col-md-auto"
            >
              <div
                className="float-end"
              >
                <span>
                  <label
                    className="form col-form-label form-display"
                  >
                    Order by:
                  </label>
                </span>
              </div>
            </div>
            <div
              className="col-md-auto"
            >
              <div
                className="float-end"
              >
                <div>
                  <select
                    className="form-select"
                    onChange={[Function]}
                    value="title"
                  >
                    <option
                      className="dropdown-item"
                      value="title"
                    >
                      Title
                    </option>
                    <option
                      className="dropdown-item"
                      value="description"
                    >
                      Description
                    </option>
                    <option
                      className="dropdown-item"
                      value="release_date"
                    >
                      Release date
                    </option>
                    <option
                      className="dropdown-item"
                      value="genre"
                    >
                      Genre
                    </option>
                    <option
                      className="dropdown-item"
                      value="platform"
                    >
                      Platform
                    </option>
                    <option
                      className="dropdown-item"
                      value="review"
                    >
                      Review
                    </option>
                  </select>
                </div>
              </div>
            </div>
            <div
              className="col-md-auto"
            >
              <div
                className="float-end"
              >
                <div>
                  <select
                    className="form-select"
                    onChange={[Function]}
                    value="1"
                  >
                    <option
                      className="dropdown-item"
                      value="1"
                    >
                      Ascending
                    </option>
                    <option
                      className="dropdown-item"
                      value="-1"
                    >
                      Descending
                    </option>
                  </select>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
